src/Ubl.php

```diff
@@ -0,0 +1,144 @@
+
+use DateTime;
+use Einvoicing\Delivery;
+use Einvoicing\Identifier;
+use Einvoicing\Invoice;
+use Einvoicing\InvoiceLine;
+use Einvoicing\Party;
+use Einvoicing\Payments\Payment;
+use Einvoicing\Payments\Transfer;
+use Einvoicing\Presets\{Peppol, Nlcius, CiusRo, CiusIt, CiusEsFace, CiusAtGov, CiusAtNat};
+use Einvoicing\Writers\UblWriter;
+use SimpleXMLElement;
+
+class Ubl extends XmlGenerator {
+
+    public const CIUS_AT_NAT = CiusAtNat::class;
+    public const CIUS_AT_GOV = CiusAtGov::class;
+    public const CIUS_ES_FACE = CiusEsFace::class;
+    public const CIUS_IT = CiusIt::class;
+    public const CIUS_RO = CiusRo::class;
+    public const NLCIUS = Nlcius::class;
+    public const PEPPOL = Peppol::class;
+
+    /** @var Invoice */
+    public $invoice;
+    /** @var Delivery */
+    public $delivery;
+
+    public function validate(string $xml, $schematron)
+    {
+        $this->invoice->validate();
+    }
+
+    public function initDocument($invoiceId, DateTime $issueDateTime, $invoiceType, ?DateTime $deliveryDate = null)
+    {
+        $this->invoice = new Invoice($this->profile);
+        $this->invoice->setNumber($invoiceId);
+        $this->invoice->setIssueDate($issueDateTime);
+        $this->invoice->setType($invoiceType->value);
+
+        $payment = new Payment();
+        $this->invoice->setPayment($payment);
+
+        $this->delivery = new Delivery();
+        if (isset($deliveryDate)) {
+            $this->hasDelivery = true;
+            $this->delivery->setDate($deliveryDate);
+        }
+        $this->invoice->setDelivery($this->delivery);
+    }
+
+    public function setSeller(string $id, InternationalCodeDesignator $idType, string $name, $tradingName = null)
+    {
+        $this->seller = new Party();
+        $this->seller->setCompanyId(new Identifier($idType->value, $id));
+        $this->seller->setName($name);
+        $this->seller->setTradingName($tradingName);
+
+        $this->invoice->setSeller($this->seller);
+    }
+
+    public function setSellerContact(?string $personName = null, ?string $telephone = null, ?string $email = null, ?string $departmentName = null)
+    {
+        $this->seller->setContactName($personName);
+        $this->seller->setContactPhone($telephone);
+        $this->seller->setContactEmail($email);
+    }
+
+    public function setSellerAddress(string $lineOne, string $postCode, string $city, string $countryCode, ?string $lineTwo = null, ?string $lineThree = null)
+    {
+        $this->seller->setAddress([$lineOne, $lineTwo, $lineThree]);
+        $this->seller->setCity($city);
+        $this->seller->setCountry($countryCode);
+        $this->seller->setPostalCode($postCode);
+    }
+
+    public function setSellerTaxRegistration(string $id, string $schemeID)
+    {
+        $this->seller->setTaxRegistrationId(new Identifier($id));
+    }
+
+    public function setBuyer(string $buyerReference, string $name, ?string $id = null)
+    {
+        $this->buyer = new Party();
```

Comment:
consider adding, in php8.2, it is deprecated to dynamically define properties
` public Party $buyer`